notebooks/MAST/HSC/CASJOBS/HCV_CASJOBS/HCV_casjobs_demo.ipynb

```diff
@@ -0,0 +1,995 @@
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# set universal matplotlib parameters\n",
+    "plt.rcParams.update({'font.size': 16})"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "HSCContext = \"HSCv3\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Set up Casjobs environment."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import getpass\n",
+    "if not os.environ.get('CASJOBS_USERID'):\n",
+    "    os.environ['CASJOBS_USERID'] = input('Enter Casjobs UserID:')\n",
+    "if not os.environ.get('CASJOBS_PW'):\n",
+    "    os.environ['CASJOBS_PW'] = getpass.getpass('Enter Casjobs password:')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### MAST Functions\n",
+    "\n",
+    "The `resolve(name)` function uses the MAST Name Resolver (which relies on both SIMBAD and NED) to get the RA, Dec position for an object."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def resolve(name):\n",
+    "    \"\"\"Get the RA and Dec for an object using the MAST name resolver\n",
+    "    \n",
+    "    Parameters\n",
+    "    ----------\n",
+    "    name (str): Name of object\n",
+    "\n",
+    "    Returns RA, Dec tuple with position\n",
+    "    \"\"\"\n",
+    "\n",
+    "    resolverRequest = {'service': 'Mast.Name.Lookup',\n",
+    "                       'params': {'input': name,\n",
+    "                                  'format': 'json'}}\n",
+    "    resolvedObjectString = mastQuery(resolverRequest)\n",
+    "    resolvedObject = json.loads(resolvedObjectString)\n",
+    "    # The resolver returns a variety of information about the resolved object, \n",
+    "    # however for our purposes all we need are the RA and Dec\n",
+    "    try:\n",
+    "        objRa = resolvedObject['resolvedCoordinate'][0]['ra']\n",
+    "        objDec = resolvedObject['resolvedCoordinate'][0]['decl']\n",
+    "    except IndexError:\n",
+    "        raise ValueError(f\"Unknown object '{name}'\")\n",
+    "    return (objRa, objDec)\n",
+    "\n",
+    "\n",
+    "def mastQuery(request, url='https://mast.stsci.edu/api/v0/invoke'):\n",
+    "    \"\"\"Perform a MAST query.\n",
+    "\n",
+    "    Parameters\n",
+    "    ----------\n",
+    "    request (dictionary): The MAST request json object\n",
+    "    url (string): The service URL\n",
+    "\n",
+    "    Returns the returned data content\n",
+    "    \"\"\"\n",
+    "    \n",
+    "    # Encoding the request as a json string\n",
+    "    requestString = json.dumps(request)\n",
+    "    r = requests.post(url, data={'request': requestString})\n",
+    "    r.raise_for_status()\n",
+    "    return r.text"
+   ]
```

Comment:
Similarly, I don't think this is needed. It's only used once, and it's ok to expect users to know to use requests!

Reply:
Deleted